File: ObjectProgramming/C++/Roslina.cpp
#include "Roslina.h"

Roslina::Roslina()
{

}

void Roslina::akcja()
{
    int szansza = rand() % 100;
    if(szansza >= MIN_SZANSZA)
    {
        int kierunek_X = rand() % 3 - 1;
        int kierunek_Y = rand() % 3 - 1;
        if(swiat->swiat_tablica[polozenie.x + kierunek_X][polozenie.y + kierunek_Y] == NULL)
        {
            swiat->nowe_stworzenie(polozenie.x + kierunek_X, polozenie.y + kierunek_Y, id);
        }
    }
}

File: ObjectProgramming/Python/Roslina.py
from Organizm import Organizm

import random

class Roslina(Organizm):
    def kolor(self):
        pass

    def akcja(self):
        a = self.polozenieX + random.randint(-1,1)
        b = self.polozenieY + random.randint(-1,1)
        if random.randint(0, 100) > self.szansza_na_rozmnozenie() and self.swiat.wolne_pole(a, b):
            if a >= 0 and a < self.swiat.wymiarX and b >= 0 and b < self.swiat.wymiarY:
                nowestworzenie = type(self)(self.swiat, a, b)
                self.swiat.organizmy.append(nowestworzenie)

    def czy_roslina(self):
        return True

    def czy_zwierze(self):
        return False

    def czy_czlowiek(self):
        return False

    def szansza_na_rozmnozenie(self):
        return 90


